src/main/java/core/basesyntax/service/handler/PurchaseHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.service.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void handleTransaction(FruitTransaction transaction) {
+        int result = Storage.fruits.get(transaction.getFruit()) - transaction.getQuantity();
+        if (result >= 0) {
+            Storage.fruits.put(transaction.getFruit(), result);

```

Comment:
This seems like you're trying to alter the constant which is illegal. After applying the singleton pattern to `Storage` class, you should be able to access the collection with: `Storage.getInstance().addFruit(...)`